File: java/src/queueing/Exp.java
package queueing ;

/**
 * Exponential distribution class. 
 * Copied from the Simulation and Modelling course toolkit.
 */
public class Exp  {
  private double rate ;

  public Exp( double r ) {
    rate = r ;
  }

  public double next() {
    return -Math.log( Math.random() ) / rate ;
  }

  public static double exp( double lam ) {
    return -Math.log( Math.random() ) / lam ;
  }
  
  public static void main(String[] args) {
	  Exp ex = new Exp(Double.parseDouble(args[0]));
	  
	  long sum = 0;
	  int iterations  = Integer.parseInt(args[1]);
	  for (int i =0; i<iterations;i++) {
		  sum += (long)ex.next();
	  }
	  
	  sum /= iterations;
	  System.out.println("Average: " + (sum));
	  
	  
  }
  
}

File: src/cinqs/tools/Exp.cpp
#include "tools/Exp.h"

#include <iostream>
#include <sstream>

Exp::Exp( double r ) {
	rate = r ;
}

const char *Exp::getDistributionSampleType() {
	std::stringstream str;
	str << "Exponential distribution with rate " << rate;
	return str.str().c_str();
}

double Exp::next() {
	return -log( ((double)rand()/(double)RAND_MAX) ) / rate ;
}

double Exp::exp( double lam ) {
	return -log( ((double)rand()/(double)RAND_MAX) ) / lam ;
}
